pkg/apis/enricher/framework/java/openliberty_detector_test.go

```diff
@@ -0,0 +1,67 @@
+package enricher
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/devfile/alizer/pkg/apis/model"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestOpenLibertyDetector_DoPortsDetection(t *testing.T) {
+	tempDir, err := os.MkdirTemp("", "openliberty-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tempDir)
+
+	hardcodedXML := `
+	<server>
+		<httpEndpoint host="*" httpPort="1234" httpsPort="1235" id="defaultHttpEndpoint"/>
+	</server>
+	`
+	writeTestFile(t, tempDir, "hardcoded_server.xml", hardcodedXML)
+
+	variableXML := `
+	<server>
+		<variable name="default.http.port" defaultValue="9080"/>
+		<variable name="default.https.port" defaultValue="9443"/>
+		<httpEndpoint host="*" httpPort="${default.http.port}" httpsPort="${default.https.port}" id="defaultHttpEndpoint"/>
+	</server>
+	`
+	writeTestFile(t, tempDir, "variable_server.xml", variableXML)
+
+	detector := OpenLibertyDetector{}
+
+	ctx := context.TODO()
+
+	t.Run("Hardcoded Ports", func(t *testing.T) {
```

Comment:
Is it possible to add a test case for a mix of hardcoded and variable based ports?

Reply:
Possibly one for empty var as well?